src/drt/src/gc/FlexGC_main.cpp

```diff
@@ -905,6 +911,192 @@ void FlexGCWorker::Impl::checkMetalSpacing()
   }
 }
 
+void FlexGCWorker::Impl::checkMetalSpacing_wrongDir_getQueryBox(gcSegment* edge,
+                                                                frCoord spcVal,
+                                                                box_t& queryBox)
+{
+  switch (edge->getDir()) {
+    case frDirEnum::W:
+      bg::set<bg::min_corner, 0>(queryBox, edge->high().x());
+      bg::set<bg::min_corner, 1>(queryBox, edge->high().y());
+      bg::set<bg::max_corner, 0>(queryBox, edge->low().x());
+      bg::set<bg::max_corner, 1>(queryBox, edge->low().y() + spcVal);
+      break;
+    case frDirEnum::E:
+      bg::set<bg::min_corner, 0>(queryBox, edge->low().x());
+      bg::set<bg::min_corner, 1>(queryBox, edge->low().y() - spcVal);
+      bg::set<bg::max_corner, 0>(queryBox, edge->high().x());
+      bg::set<bg::max_corner, 1>(queryBox, edge->high().y());
+      break;
+    case frDirEnum::S:
+      bg::set<bg::min_corner, 0>(queryBox, edge->high().x() - spcVal);
+      bg::set<bg::min_corner, 1>(queryBox, edge->high().y());
+      bg::set<bg::max_corner, 0>(queryBox, edge->low().x());
+      bg::set<bg::max_corner, 1>(queryBox, edge->low().y());
+      break;
+    case frDirEnum::N:
+      bg::set<bg::min_corner, 0>(queryBox, edge->low().x());
+      bg::set<bg::min_corner, 1>(queryBox, edge->low().y());
+      bg::set<bg::max_corner, 0>(queryBox, edge->high().x() + spcVal);
+      bg::set<bg::max_corner, 1>(queryBox, edge->high().y());
+      break;
+    default:
+      break;
+  }
+}
+
+void FlexGCWorker::Impl::checkMetalSpacing_wrongDir(gcPin* pin, frLayer* layer)
+{
+  auto lef58WrongDirCons = layer->getLef58SpacingWrongDirConstraints();
+  auto layerNum = layer->getLayerNum();
+  for (auto con : lef58WrongDirCons) {
+    if (con->hasLength()) {
+      logger_->warn(
+          DRT, 333, "Unsupported branch LENGTH in checkMetalSpacing_wrongDir.");
+    }
+    if (con->hasPrlLength()) {
+      auto prlLength = con->getPrlLength();
+      if (prlLength < 0) {
+        logger_->warn(DRT,
+                      334,
+                      "Unsupported negative value for prlLength in branch PRL "
+                      "in checkMetalSpacing_wrongDir.");
+      }
+    }
```

Comment:
Could you move these checks to IO so that we don't get a warning each time we do a gc check?